lib/galaxy/tools/actions/__init__.py

```diff
@@ -51,12 +51,12 @@ class ToolAction:
     been converted and validated).
     """
 
-    def execute(self, tool, trans, incoming=None, set_output_hid=True):
+    def execute(self, tool, trans, incoming=None, set_output_hid=True, **kwargs):
         incoming = incoming or {}
```

Comment:
```suggestion
```

Also, should this be decorated `@abstractmethod` instead of raising `TypeError` ?

Reply:
If I add `@abstractmethod` and inherit from `ABCMeta` then:
```
lib/galaxy/tools/__init__.py:993: error: All overload variants require at least one argument  [call-overload]
lib/galaxy/tools/__init__.py:993: note: Possible overload variants:
lib/galaxy/tools/__init__.py:993: note:     def (o: object) -> DefaultToolAction
lib/galaxy/tools/__init__.py:993: note:     def (name: str, bases: Tuple[type, ...], dict: Dict[str, Any], **kwds: Any) -> DefaultToolAction
test/unit/app/tools/test_actions.py:72: error: All overload variants of "DefaultToolAction" require at least one argument  [call-overload]
test/unit/app/tools/test_actions.py:72: note: Possible overload variants:
test/unit/app/tools/test_actions.py:72: note:     def DefaultToolAction(self, o: object) -> DefaultToolAction
test/unit/app/tools/test_actions.py:72: note:     def DefaultToolAction(self, name: str, bases: Tuple[type, ...], dict: Dict[str, Any], **kwds: Any) -> DefaultToolAction

```